server/monitor/src/main/java/org/apache/accumulo/monitor/rest/api/TabletServerInformation.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.monitor.rest.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.bind.annotation.XmlAttribute;
+
+import org.apache.accumulo.core.master.thrift.RecoveryStatus;
+import org.apache.accumulo.core.master.thrift.TableInfo;
+import org.apache.accumulo.core.master.thrift.TabletServerStatus;
+import org.apache.accumulo.monitor.Monitor;
+import org.apache.accumulo.server.util.TableInfoUtil;
+
+public class TabletServerInformation {
+
+  @XmlAttribute(name = "id")
+  public String server;
+
+  public String hostname;
+  public long lastContact;
+  public double osload;
+
+  public CompactionsTypes compactions;
+
+  public int tablets;
+  public double ingest, query, ingestMB, queryMB;
+  public Integer scans; // For backwards compatibility, has same information as scansRunning
+  public Double scansessions;
+  public Double scanssessions; // For backwards compatibility
+  public long holdtime;
+
+  // New variables
+
+  public String ip;
+  private Integer scansRunning, scansQueued, minorRunning, minorQueued, majorRunning, majorQueued;
+  private CompactionsList scansCompacting, major, minor; // if scans is removed, change scansCompacting to scans
+  public long entries, lookups, indexCacheHits, indexCacheRequests, dataCacheHits, dataCacheRequests;
+  public double indexCacheHitRate, dataCacheHitRate;
+  public List<RecoveryStatusInformation> logRecoveries;
+
+  public TabletServerInformation() {}
+
+  public TabletServerInformation(TabletServerStatus thriftStatus) {
+    TableInfo summary = TableInfoUtil.summarizeTableStats(thriftStatus);
+    updateTabletServerInfo(thriftStatus, summary);
+  }
+
+  public void updateTabletServerInfo(TabletServerStatus thriftStatus, TableInfo summary) {
+
+    long now = System.currentTimeMillis();
+
+    this.server = this.ip = this.hostname = thriftStatus.name;
+    this.tablets = summary.tablets;
```

Comment:
Maybe a check that TableInfo is non-null before using it the first time? `Objects.requireNonNull(summary).tablets` would be an easy modification.